File: examples/sample.py
# Sample Python file for ReviewerBot testing

def calculate_average(numbers):
    """Calculate the average of a list of numbers."""
    if not numbers:
        return 0
    return sum(numbers) / len(numbers)

def validate_email(email):
    """Validate email format."""
    return '@' in email and '.' in email

def process_text(text):
    """Process and clean text input."""
    return text.strip().lower()

def get_user_info(user_id):
    """Retrieve user information."""
    return {
        'id': user_id,
        'name': 'Alice Johnson',
        'age': 25
    }

def complex_calculation(x, y, z):
    """Perform complex mathematical operations."""
    result = (x ** 2 + y ** 2) / z
    return abs(result) 

File: examples/sample.go
package main

import (
	"strings"
)

// CalculateSum adds two numbers together
func CalculateSum(a, b int) int {
	return a + b
}

// ProcessString converts a string to uppercase
func ProcessString(input string) string {
	return strings.ToUpper(input)
}

// ValidateEmail checks if an email is valid
func ValidateEmail(email string) bool {
	return strings.Contains(email, "@") && strings.Contains(email, ".")
}

// GetUserData retrieves user information
func GetUserData(userID string) map[string]interface{} {
	return map[string]interface{}{
		"id":   userID,
		"name": "John Doe",
		"age":  30,
	}
}

// ComplexCalculation performs a complex mathematical operation
func ComplexCalculation(x, y, z float64) float64 {
	result := x * y / z
	if result < 0 {
		result = -result
	}
	return result
}
